src/ThriveWorkerUnit.sol

```diff
@@ -0,0 +1,149 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import "./IBadgeQuery.sol";
+
+/**
+ * @title ThriveWorkerUnit
+ * @dev Contract of a work unit representing a task where contributors complete and earn rewards.
+ */
+contract ThriveWorkerUnit {
+    using EnumerableSet for EnumerableSet.AddressSet;
+    using EnumerableSet for EnumerableSet.Bytes32Set;
+
+    address public immutable moderator;
+    address public immutable rewardToken;
+    uint256 public immutable rewardAmount;
+    uint256 public immutable maxRewards;
+    uint256 public immutable validationRewardAmount;
+    uint256 public maxCompletionsPerUser;
+    uint256 public deadline;
```

Comment:
Assigned Address is missing, it can be zero address if no address is needed. Also would be good to have a function like this:
```
function status() external view returns (string memory) {
  return block.timestamp <= deadline ? 'active' : 'expired';
}
```